render-app/src/main/java/org/janelia/alignment/match/OrderedCanvasIdPair.java

```diff
@@ -55,6 +57,47 @@ public OrderedCanvasIdPair(final CanvasId oneId,
         this.absoluteDeltaZ = deltaZ == null ? null : Math.abs(deltaZ);
     }
 
+    /**
+     * @param  oneTileBounds      identifiers and bounds for one canvas identifier.
+     * @param  anotherTileBounds  identifiers and bounds for another canvas identifier.
+     *
+     * @return an ordered pair where each CanvasId includes relative position information
+     *         based upon the specified tile bounds.
+     *         Assumes that the bounds are for the same layer so the pair's deltaZ is set to null.
+     *
+     * @throws IllegalArgumentException
+     *   if both tile identifiers are the same.
+     */
+    public static OrderedCanvasIdPair withRelativePositions(final TileBounds oneTileBounds,
+                                                            final TileBounds anotherTileBounds)
+            throws IllegalArgumentException {
+
+        final CanvasId oneCanvasId = new CanvasId(oneTileBounds.getSectionId(), oneTileBounds.getTileId());
+        final CanvasId anotherCanvasId = new CanvasId(anotherTileBounds.getSectionId(), anotherTileBounds.getTileId());
+
+        final MontageRelativePosition[] positions = MontageRelativePosition.getRelativePositions(oneTileBounds,
+                                                                                                 anotherTileBounds);
+        final int comparisonResult = oneCanvasId.compareTo(anotherCanvasId);
+        // if the tile identifiers are not in natural order ...
+        if (comparisonResult > 0) {
+            final MontageRelativePosition[] flippedPositions =
+                    MontageRelativePosition.getRelativePositions(anotherTileBounds,
+                                                                 oneTileBounds);
+            // and if both tiles are in the same position relative to each other ...
+            if (positions[0].equals(flippedPositions[0])) {
+                // then flip the positions to maintain consistency with natural order
+                positions[0] = flippedPositions[1]; // NOTE: this is flipping because both are same
+                positions[1] = flippedPositions[0];
```

Comment:
I see that swapping `positions` without an auxiliary variable is not possible here and assigning from `flippedPositions` is convenient. However, from the naming it seems at first glance that this does **not** flip the array items.

Reply:
tried to improve naming